lib/token-store.js

```diff
@@ -8,196 +8,1110 @@
 // Implementation
 //------------------------------------------------------------------------------
 
-module.exports = function(tokens) {
-    const api = {},
-        starts = Object.create(null),
-        ends = Object.create(null),
-        length = tokens.length;
+const W_INIT = -3;
+const W_TOKENS = 1;
+const W_COMMENTS = 2;
```

Comment:
Those are used `TokenCommentCursor#which` property; the flag which indicates the cursor is using which index. I will add comments.